cpp/include/cudf/strings/combine.hpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/column/column.hpp>
+
+namespace cudf
+{
+namespace strings
+{
+
+/**
+ * @brief Row-wise concatenates the given list of strings columns and
+ * returns a single strings column result.
+ *
+ * Each new string is created by concatenating the strings from the same
+ * row delimited by the separator provided.
+ *
+ * Any row with a null entry will result in the corresponding output
+ * row to be null entry unless a narep string is specified to be used
+ * in its place.
+ *
+ * The number of strings in the columns provided must be the same.
+ *
+ * ```
+ * s1 = ['aa', null, '', 'aa']
+ * s2 = ['', 'bb', 'bb', null]
+ * r1 = concatenate([s1,s2])
+ * r1 is ['aa', null, 'bb', null]
+ * r2 = concatenate([s1,s2],':','_')
+ * r2 is ['aa:', '_:bb', ':bb', 'aa:_']
+ * ```
+ *
+ * @throw cudf::logic_error if input columns are not all the same size.
+ *
+ * @param strings_columns List of string columns to concatenate.
+ * @param separator String that should inserted between each string from each row.
+ *        Default is an empty string.
+ * @param narep Null-terminated CPU string that should be used in place of any null strings
+ *        found in any column. Default of null means any null entry in any column will
+ *        produces a null result for that row.
+ * @param mr Resource for allocating device memory.
+ * @return New column with concatenated results.
+ */
+std::unique_ptr<cudf::column> concatenate( const std::vector<strings_column_view>& strings_columns,
```

Comment:
How about this? Makes it clear that all columns are expected to have the same number of elements. 
```suggestion
std::unique_ptr<cudf::column> concatenate( table_view const& strings_columns,
```

Reply:
I cannot put a vector of `strings_column_views` inside of a `table_view'

Reply:
No, but you can do a `std:vector<column_view>`. And `strings_column_view`s is trivially convertible to a `column_view`. 

`table` and `table_view` exist for a reason to represent a set of columns of the same length.